### ISSUE TITLE
Duplications between Rnafusion and Taxprofiler

### ISSUE THREAD

**Issue by sofstam** [Clinical-Genomics/cg#2044]
## Description

From @ivadym comments in #1998 
There are already some duplications betweeen rnafusion and taxprofiler and this needs to be adressed as soon as possible. It is not sustainable in the long term to have methods with the same names and that share the same purpose and code.

## Suggested solution

If we're having this as a summer project, as we discussed, it would be nice to have a refinement meeting beforehand to decide how we include nextflow "analysis" APIs in meta:

The current NextflowAnalysisAPI is a handler and probably belongs to utils and not to meta.
We need a common class for taxprofiler and rnafusion APIs to avoid code duplication and increase in complexity. And decide if it should take analysis API as a parent. Probably some of the current NextflowAnalysisAPI methods could be moved here.

## This can be closed when

- [ ] We have had a refinement meeting
- [ ] The issues above have been adressed

## Blocked by

*If there are any blocking issues/prs/things in this or other repos. Please link to them.*

**Comment by fevac** [comment]
This is related to this https://github.com/Clinical-Genomics/cg/issues/1819

**Comment by fevac** [comment]
Closing this issue as the most relevant parts are done. I will open separate issues for what is left to do
